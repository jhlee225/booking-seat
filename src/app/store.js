import { configureStore } from "@reduxjs/toolkit";
import loginReducer from "../features/login/loginSlice";
import seatReducer from "../features/seat/seatSlice";
import menuReducer from "../features/menu/menuSlice";
import bookReducer from "../features/booking/BookSlice";
import timeReducer from "../features/time/timeSlice";
import signUpReducer from "../features/signup/signupSlice";

export default configureStore({
  reducer: {
    login: loginReducer,
    seat: seatReducer,
    menu: menuReducer,
    time: timeReducer,
    book: bookReducer,
    signUp: signUpReducer,
  },
});
