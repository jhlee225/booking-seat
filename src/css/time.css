.times {
  border-radius: 10px;
  padding: 20px 5px;
  display: flex;
  background-color: rgb(245, 245, 245, 0.9);
  width: 75vmax;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}
