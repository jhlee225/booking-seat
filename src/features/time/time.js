import React from "react";
import { Link } from "react-router-dom";
export function Time() {
  return (
    <>
      <div className="times">
        <input type="date" />
        <BlankRow />
        <div>현재 예약 현황</div>
        <Link to="/Book">
          <div className="btn">예약하기</div>
        </Link>
        <BlankRow />
      </div>
    </>
  );
}

function BlankRow() {
  return (
    <div className="row">
      <div className="blank"></div>
      <div className="blank"></div>
      <div className="blank"></div>
      <div className="blank"></div>
      <div className="blank"></div>
      <div className="blank"></div>
      <div className="blank"></div>
      <div className="blank"></div>
      <div className="blank"></div>
      <div className="blank"></div>
      <div className="blank"></div>
      <div className="blank"></div>
      <div className="blank"></div>
      <div className="blank"></div>
    </div>
  );
}
