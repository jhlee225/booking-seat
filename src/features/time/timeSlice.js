import { createSlice } from "@reduxjs/toolkit";

export const seatSlice = createSlice({
  name: "seat",
  initialState: {
    seat: "",
    time: [],
  },
  reducers: {
    setFloor: (state, action) => {
      state.floor = action.payload.floor;
    },
    setSeats: (state, action) => {
      state.seats = action.payload.seats;
    },
  },
});

export const { setFloor, setSeats } = seatSlice.actions;

export const selectFloor = (state) => state.seat.floor;
export const selectSeats = (state) => state.seat.seats;

export default seatSlice.reducer;
