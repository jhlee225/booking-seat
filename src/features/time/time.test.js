import React from "react";
import { render } from "@testing-library/react";
import { HashRouter as Router, Route } from "react-router-dom";
import { Provider } from "react-redux";
import store from "../../app/store";
import { Time } from "./time";

it("render Seats", () => {
  const seatsRendered = render(
    <Provider store={store}>
      <Router>
        <Time />
      </Router>
    </Provider>
  );
  expect(seatsRendered.container).toMatchSnapshot();
});

describe("make graph", () => {
  const seatsRendered = render(
    <Provider store={store}>
      <Router>
        <Time />
      </Router>
    </Provider>
  );
});
