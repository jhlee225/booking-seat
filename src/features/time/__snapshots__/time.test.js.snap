// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Seats 1`] = `
<div>
  <div
    class="times"
  >
    <div>
      번 자리
    </div>
    <div>
      현재 예약 현황
    </div>
    <div>
      <span>
        날짜 : 
      </span>
      <input
        type="date"
      />
    </div>
    <div
      class="btns"
    >
      <a
        href="#/Seat"
      >
        <div
          class="back btn"
        >
          뒤로가기
        </div>
      </a>
      <a
        href="#/Book"
      >
        <div
          class="btn"
        >
          예약하기
        </div>
      </a>
    </div>
  </div>
</div>
`;
